### ISSUE TITLE
Bug: react/prop-types rule failed when using @babel/eslint-parser@7.16.0

### ISSUE THREAD

**Issue by lyz810** [eslint/eslint#15237]
### Environment

Environment Info:

Node version: v16.12.0
npm version: v8.1.0
Local ESLint version: v8.1.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 20.6.0

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
// ...
 extends: ['eslint-config-airbnb'],
  parser: '@babel/eslint-parser',
  parserOptions: {
    babelOptions: {
      configFile: path.join(__dirname, './babel.config.js'),
    },
  },
// ...
}
```
</details>

```jsx
import React from 'react';
import PropTypes from 'prop-types';

class A extends React.PureComponent {
  static propTypes = {
    test: PropTypes.string.isRequired,
  };

  render() {
    const { test } = this.props;
    return (<div>{test}</div>);
  }
}

export default A;
```

### What did you expect to happen?

no eslint error

### What actually happened?

error  'test' is missing in props validation  react/prop-types

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

This error appears when I upgrade @babel/eslint-parser from v7.15.8 to v7.16.0
I don't know which npm package causes this bug, so I report it in both eslint an @babel/eslint-parser
You can find more information at follow link:
[https://github.com/babel/babel/issues/13906](url)

**Comment by aladdin-add** [comment]
related: https://github.com/yannickcr/eslint-plugin-react/issues/3055

`eslint-plugin-react` does not support eslint v8 yet. please keep using eslint v7 util it fully supports.

**Comment by nzakas** [comment]
Closing as this is not an issue with ESLint.
